src/DocumentFragment.js

```diff
@@ -0,0 +1,71 @@
+  // ### DocumentFragment Shim
+  //
+  // A representation of a document fragment that can be normally generated
+  // by calling `document.createDocumentFragment`. Since the document fragment
+  // is itself a node, it inherits all properties of a node.
+module.exports = (function(){
+  "use strict";
+
+  var Node = require('./Node');
+  var voidHtmlElements = ["area", "base", "br", "col", "command", "embed", "hr",
+                          "img", "input", "keygen", "link", "meta", "param",
+                          "source", "track", "wbr"];
+
+  function assertExists(name, value) {
+    if(!value || value === "") {
+      throw new Error("Expected a value for "  + name);
+    }
+
+    return true;
+  }
+
+  function initNode(node, type, name) {
+    Object.defineProperties(node, {
+      "nodeType": {
+        get: function() { return type; }
+      },
+      "nodeName": {
+        get: function() { return name; }
+      }
+    });
+
+    return node;
+  }
+
+  function DocumentFragment() {
+    var node = this.node = new Node();
+    node = initNode(node, node.DOCUMENT_FRAGMENT_NODE, "#document-fragment");
+    node._ownerDocument = this;
+  }
+
+  DocumentFragment.prototype.createTextNode = function(data) {
+    var node = new Node(data);
+    return initNode(node, node.TEXT_NODE, "#text");
+  };
+
+  DocumentFragment.prototype.createElementNS = function(namespaceURI, qualifiedName) {
+    var node = new Node(false, namespaceURI);
+    var nodeName = assertExists("nodeName", qualifiedName) && qualifiedName.toUpperCase();
+    if(voidHtmlElements.indexOf(nodeName.toLowerCase()) > -1) {
+      node.isVoid = true;
+    }
+    return initNode(node, node.ELEMENT_NODE, nodeName);
+  };
+
+  DocumentFragment.prototype.createElement = function(tagName) {
+    return this.createElementNS(false, tagName);
+  };
+
+  DocumentFragment.prototype.createComment = function(data) {
+    var node = new Node(data);
+    return initNode(node, node.COMMENT_NODE, "#comment");
+  };
+
+  DocumentFragment.prototype.createAttribute = function(name) {
+    var node = new Node();
+    var nodeName = assertExists("nodeName", name) && name.toLowerCase();
```

Comment:
This is "doing more than the spec says you should". Empty attributes are entirely fine, don't restrict what we can do unless you're absolutely sure, with spec backing, that it must be done.